src/NodeAnalyzer/CompactFuncCallAnalyzer.php

```diff
@@ -27,13 +28,32 @@ public function isInCompact(FuncCall $funcCall, Variable $variable): bool
             return false;
         }
 
-        $args = $funcCall->args;
-        foreach ($args as $arg) {
-            if (! $arg->value instanceof String_) {
+        return $this->isInArgOrArrayItemNodes($funcCall->args, $variableName);
+    }
+
+    /**
+     * @param Array<\PhpParser\Node\Arg|\PhpParser\Node\Expr\ArrayItem|null> $nodes
+     */
+    private function isInArgOrArrayItemNodes(array $nodes, string $variableName): bool
+    {
+        foreach ($nodes as $node) {
+            if ($node === null) {
+                continue;
+            }
+
+            if ($node->value instanceof Array_) {
+                if ($this->isInArgOrArrayItemNodes($node->value->items, $variableName)) {
+                    return true;
+                }
+
+                continue;
+            }
+
+            if (! $node->value instanceof String_) {
                 continue;
             }
 
-            if ($arg->value->value === $variableName) {
+            if ($node->value->value === $variableName) {
                 return true;
             }
```

Comment:
can return directly:

```suggestion
            return $node->value->value === $variableName;
```